### PR TITLE
Copy over 0.4.2 from react-dfp pkg

### PR DESCRIPTION
**Pull request by wflann**
[Story](https://rentpath.leankit.com/card/570822877)

The previous version of this library that we forked did not include support for react v16. I copied over the library to incorporate the changes needed for the upgrade and support on rent-js.

### CODE REVIEW

File: js/manager.js [hunk 1]
```diff
@@ -81,41 +82,44 @@ const DFPManager = Object.assign(new EventEmitter().setMaxListeners(0), {
     availableSlots = Object.keys(availableSlots)
       .filter(id => !availableSlots[id].loading)
       .reduce(
-      (result, id) => Object.assign(
-        result, {
-          [id]: Object.assign(availableSlots[id], { loading: true }),
-        },
-      ), {},
-    );
+        (result, id) => Object.assign(
```
Comment (tadjohnston) [inline js/manager.js]: Let's keep the indentation what it is in the lib if we can.

File: js/manager.js [hunk 2]
```diff
@@ -161,15 +165,15 @@ const DFPManager = Object.assign(new EventEmitter().setMaxListeners(0), {
   },
 
   registerSlot({
```
Comment (tadjohnston) [inline js/manager.js]: Same with indentation here.

File: js/manager.js (hunk 1 above)
Comment (wflann) [reply js/manager.js]: Good call. Updated

**Comment by danthamotheram** [comment]
I'm 👍 with this if Tad is, with the understanding that if AdOps gives us the go-ahead, we can use the original React DFP dependency directly. And hopefully we'll get a chance to replace that in time, with our own GPT API wrapper. I'll create a story for that.

**Comment by tadjohnston** [comment]
RTM 👍

**Pull request merged**